core/state_transition.go

```diff
@@ -302,10 +308,9 @@ func (st *StateTransition) preCheck() error {
 	}
 	if !msg.SkipFromEOACheck {
 		// Make sure the sender is an EOA
-		codeHash := st.state.GetCodeHash(msg.From)
-		if codeHash != (common.Hash{}) && codeHash != types.EmptyCodeHash {
-			return fmt.Errorf("%w: address %v, codehash: %s", ErrSenderNoEOA,
-				msg.From.Hex(), codeHash)
+		code := st.state.GetCode(msg.From)
+		if len(code) > 0 && !bytes.HasPrefix(code, types.DelegationPrefix) {
```

Comment:
This checks only for the delegation prefix, while most other code uses `types.ParseDelegation` which checks the size as well. Maybe we should add `types.IsDelegation` and use that here and within parsedelegation to keep the checks consistent (not an issue right now as long as we only have one length delegations, but might be a problem on chains with predeployed contracts)